src/security_utilities_rust/src/microsoft_security_utilities_core/identifiable_secrets.rs

```diff
@@ -811,3 +814,114 @@ fn index_of(input_string: String, search_char: char) -> i32
     }
 }
 
+#[derive(Clone)]
+struct Detection {
+    start: u64,
+    end: u64,
+    length: usize,
+    redaction_token: String
+}
+
+pub struct SecretMasker {
+    pub scan: identifiable_scans::Scan
+}
+
+impl SecretMasker {
+    pub fn mask_secrets(&mut self, input: &str, default_redaction_token: Option<&str>, validate_checksum: bool) -> String {
```

Comment:
It's not immediate, but I would like to see a way to mask_secrets on files that are binary and not a text string. We have large binary files that could use this masking as well.

Reply:
@beaubelgrave Tracking via https://github.com/microsoft/security-utilities/issues/62